### PR TITLE
Gym profile matching : ignore profile that do not match the correct platform

### PR DESCRIPTION
**Pull request by nicolasbraun**
### Motivation and Context
When a project has an iOS and a tvOS target the bundle identifier are the same.
It should not result in a duplicate error when matching.

Fixes: #10341

This PR includes PR #10386 but as it may lead to more discussion I made 2 PRs

### What did I do
- Detect what should be the base SDK in the build_settings based on the destination option in gym
- Filter target based on this configuration

### CODE REVIEW

**Review by joshdholtz — DISMISSED** [review]
Also having some tests would be nice 😊 ❤️

File: gym/lib/gym/code_signing_mapping.rb [hunk 1]
```diff
@@ -104,6 +104,21 @@ def test_target?(build_settings)
...
+      sdkroot = ""
+      case destination
+      when "macosx"
+        sdkroot = "macosx"
+      when "iOS"
+        sdkroot = "iphoneos"
+      when "tvOS"
+        sdkroot = "appletvos"
+      end
+      return build_settings["SDKROOT"] != sdkroot
```
Comment (thasegaw) [inline gym/lib/gym/code_signing_mapping.rb]: If the `SDKROOT` is `watchos` and `destination` is `iOS`, it should be handled as same platform.
The Provisioning Profiles for watchOS is necessary for archiving iOS app.

**Review by thasegaw — CHANGES_REQUESTED** [review]

**Comment by nicolasbraun** [comment]
@joshdholtz test added + some other changes / error catches

**Comment by nicolasbraun** [comment]
@joshdholtz any chances this gets in the next release?

**Comment by nicolasbraun** [comment]
Signed, changed email

**Comment by nicolasbraun** [comment]
Test to trigger google bot :)

**Comment by joshdholtz** [comment]
@nicolasbraun Will review today! Will be getting to my computer/office in about hour and half-ish 👀

**Comment by nicolasbraun** [comment]
trigger cla

File: gym/lib/gym/code_signing_mapping.rb [hunk 2]
```diff
@@ -104,6 +104,21 @@ def test_target?(build_settings)
       return (!build_settings["TEST_TARGET_NAME"].nil? || !build_settings["TEST_HOST"].nil?)
     end
 
+    def same_platform?(sdkroot)
+      destination = Gym.config[:destination].dup
+      destination.slice! "generic/platform="
```
Comment (taquitos) [inline gym/lib/gym/code_signing_mapping.rb]: style nit, we like to use ()'s: 
`destination.slice! "generic/platform="` ->`destination.slice!("generic/platform=")`

File: gym/lib/gym/code_signing_mapping.rb [hunk 3]
```diff
@@ -104,6 +104,21 @@ def test_target?(build_settings)
...
+      destination_sdkroot = []
+      case destination
+      when "macosx"
+        destination_sdkroot = ["macosx"]
+      when "iOS"
+        destination_sdkroot = ["iphoneos", "watchos"]
+      when "tvOS"
+        destination_sdkroot = ["appletvos"]
+      end
+      return destination_sdkroot.include? sdkroot
```
Comment (taquitos) [inline gym/lib/gym/code_signing_mapping.rb]: (same here)

**Review by taquitos — APPROVED** [review]
I think this looks great. Thank you for adding that test! 🎈:goat:

**Comment by nicolasbraun** [comment]
Goatta find a smiley signature :) Thanks for 👀

**Pull request merged**